src/whitelistHandler.ts

```diff
@@ -0,0 +1,63 @@
+import * as fs from 'fs';
+import * as jsyaml from 'js-yaml';
+
+let trivyWhitelistPath = "";
+let dockleWhitelistPath = "";
+export let trivyWhitelistExists = false;
+
+export function getTrivyWhitelist(): string {
+    if (trivyWhitelistExists)
+        return trivyWhitelistPath;
+    else
+        throw new Error("Could not find whitelist file for common vulnerabilities");
+}
+
+function initializeTrivyWhitelistPath() {
+    let curDate = Date.now();
+    let trivyWhitelistDir = `${process.env['GITHUB_WORKSPACE']}/containerscan_${curDate}/trivy`;
+    fs.mkdirSync(trivyWhitelistDir, { recursive: true });
+    trivyWhitelistPath = trivyWhitelistDir + "/.trivyignore";
+}
+
+function initializeDockleWhitelistPath() {
+    //Dockle expects .dockleignore file at the root of the repo 
+    dockleWhitelistPath = `${process.env['GITHUB_WORKSPACE']}/.dockleignore`;
+}
+
+export function init() {
+    const whitelistFilePath = `${process.env['GITHUB_WORKSPACE']}/.github/containerscan/whitelist.yaml`;
+    if (!fs.existsSync(whitelistFilePath)) {
+        console.log("Could not find whitelist file.");
+        return;
+    }
+
+    initializeTrivyWhitelistPath();
+    initializeDockleWhitelistPath();
+
+    try {
+        const whitelistYaml = jsyaml.safeLoad(fs.readFileSync(whitelistFilePath, 'utf8'));
+        if (whitelistYaml.general) {

```

Comment:
What's `.general` for? Why are we forking it like this?

Reply:
You don't seem to be handling it for other images. Is this expected? Based on the image you've pasted in description

Reply:
For the first version of the action, we are supporting only a single list of vulnerabilities for CVEs and CISs. In later versions, we are planning to support image specific whitelisting. Keeping in mind back-compatibility and extensibility, I've added the general label where the customer can put all the vulnerabilities irrespective of the image.